test/integration/shoots/applications/shoot_app_test.go

```diff
@@ -0,0 +1,307 @@
+
+	"github.com/gardener/gardener/pkg/apis/garden/v1beta1"
+	. "github.com/gardener/gardener/test/integration/framework"
+	"github.com/sirupsen/logrus"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+)
+
+var (
+	kubeconfig        = flag.String("kubeconfig", "", "the path to the kubeconfig  of the garden cluster that will be used for integration tests")
+	seedName          = flag.String("seedName", "", "the name of the seed for the test shoot")
+	shootName         = flag.String("shootName", "", "the name of the shoot we want to test")
+	shootNamespace    = flag.String("shootNamespace", "", "the namespace name that the shoot resides in")
+	testShootsPrefix  = flag.String("prefix", "", "prefix to use for test shoots")
+	logLevel          = flag.String("verbose", "", "verbosity level, when set, logging level will be DEBUG")
+	downloadPath      = flag.String("downloadPath", "/tmp/test", "the path to which you download the kubeconfig")
+	shootTestYamlPath = flag.String("shootpath", "", "the path to the shoot yaml that will be used for testing")
+	cleanup           = flag.Bool("cleanup", false, "deletes the newly created / existing test shoot after the test suite is done")
+)
+
+const (
+	GuestbookAppTimeout         = 10 * time.Minute
+	DownloadKubeconfigTimeout   = 10 * time.Minute
+	WaitForShootCreationTimeout = 3600 * time.Second
+	WaitForShootDeletionTimeout = 3600 * time.Second
+	DashboardAvailableTimeout   = 1 * time.Minute
+	APISERVER                   = "kube-apiserver"
+	GuestBookTemplateName       = "guestbook-app.yaml.tpl"
+)
+
+var _ = Describe("Shoot application testing", func() {
+	var (
+		shootGardenerTest   *ShootGardenerTest
+		shootTestOperations *GardenerTestOperation
+		tillerOptions       *TillerOptions
+		shootAppTestLogger  *logrus.Logger
+		apiserverLabels     map[string]string
+		guestBooktpl        *template.Template
+		targetTestShoot     *v1beta1.Shoot
+	)
+
+	BeforeSuite(func() {
+		if len(*kubeconfig) == 0 {
+			Fail("The path to the Garden cluster kubeconfig is required, please specify it and try again")
+		}
+
+		shootAppTestLogger = logger.AddWriter(logger.NewLogger(*logLevel), GinkgoWriter)
+
+		tillerOptions = &TillerOptions{
+			DeploymentName:    "tiller-deploy",
+			Namespace:         metav1.NamespaceSystem,
+			EnableHostNetwork: false,
+		}
+
+		// check if a shoot spec is provided, if yes create a shoot object from it and use it for testing
+		// alternatively, if the
+		if len(*shootTestYamlPath) > 0 {
+			// parse shoot yaml into shoot object and generate random test names for shoots
+			_, shootObject, err := CreateShootTestArtifacts(*shootTestYamlPath, *testShootsPrefix)
+			Expect(err).NotTo(HaveOccurred())
+
+			shootGardenerTest, err = NewShootGardenerTest(*kubeconfig, shootObject, shootAppTestLogger)
+			Expect(err).NotTo(HaveOccurred())
+
+			timeoutCtx, cancel := context.WithTimeout(context.Background(), WaitForShootCreationTimeout)
```

Comment:
This won't work - you should wrap `BeforeSuite` as we did for for `It` and contextify your method - otherwise, your timeouts won't be recognized by Ginkgo.